third_party/move/move-binary-format/src/deserializer.rs

```diff
@@ -942,6 +974,87 @@ fn load_signature_tokens(cursor: &mut VersionedCursor) -> BinaryLoaderResult<Vec
     Ok(tokens)
 }
 
+fn load_access_specifiers(
+    cursor: &mut VersionedCursor,
+) -> BinaryLoaderResult<Option<Vec<AccessSpecifier>>> {
+    load_option(cursor, |cursor| {
+        let mut specs: Vec<AccessSpecifier> = vec![];
+        for _ in 0..load_access_specifier_count(cursor)? {
```

Comment:
This is run in the loader.  Pre-allocating the vector to the right size will save a bit of time.

Reply:
Ah! I didn't even got the point because this kind of micro-optimization is very uncommon. Sure, done.

I BTW don't think the collect would work, because the implementation does not the size of the iterator ahead of time.